tests/Fixer/Whitespace/SpacesInsideParenthesisFixerTest.php

```diff
@@ -0,0 +1,361 @@
+            ],
+            // list call with trailing comma - need to leave alone
+            [
+                '<?php list($path, $mode, ) = foo();',
+            ],
+            [
+                '<?php list($path, $mode,) = foo();',
+            ],
+            [
+                '<?php
+$a = $b->test(  // do not remove space
+    $e          // between `(` and `)`
+                // and this comment
+);',
+            ],
+        ];
+    }
+
+    public function provideSpacesFixCases()
+    {
+        return [
+            [
+                '<?php foo();',
+                '<?php foo( );',
+            ],
+            [
+                '<?php
+if ( true ) {
+    // if body
+}',
+                '<?php
+if (true) {
+    // if body
+}',
+            ],
+            [
+                '<?php
+if ( true ) {
+    // if body
+}',
+                '<?php
+if (     true   ) {
+    // if body
+}',
+            ],
+            [
+                '<?php
+function foo( $bar, $baz )
+{
+    // function body
+}',
+                '<?php
+function foo($bar, $baz)
+{
+    // function body
+}',
+            ],
+            [
+                '<?php
+$foo->bar( $arg1, $arg2 );',
+                '<?php
+$foo->bar(  $arg1, $arg2   );',
+            ],
+            [
+                '<?php
+$var = array( 1, 2, 3 );
+',
+                '<?php
+$var = array(1, 2, 3);
+',
+            ],
+            [
+                '<?php
+$var = [ 1, 2, 3 ];
+',
+            ],
+            [
+                '<?php list( $path, $mode, ) = foo();',
+                '<?php list($path, $mode,) = foo();',
+            ],
+            [
+                '<?php
+$a = $b->test(  // do not remove space
+    $e          // between `(` and `)`
+                // and this comment
+ );',
+            ],
+            [
+                '<?php
+function foo( $bar, $baz )
+{
+    // function body
+}',
+                '<?php
+function foo($bar, $baz)
+{
+    // function body
+}',
+            ],
+            [
+                '<?php
+function hello( $value ) {
+    // code...
+}',
+                '<?php
+function hello($value) {
+    // code...
+}',
+            ],
+            [
+                '<?php
+$code = function ( $hello, $there ) use ( $ami, $tumi ) {
+    // code
+};
+',
+                '<?php
+$code = function ($hello, $there) use ($ami, $tumi) {
+    // code
+};
+',
+            ],
+            [
+                '<?php
+for ( $i = 0; $i < 42; $i++ ) {
+    // code...
+}
+',
+                '<?php
+for ($i = 0; $i < 42; $i++) {
+    // code...
+}
+',
+            ],
+            [
+                '<?php
+explode( $a, $b );
+',
+                '<?php
+explode($a, $b);
+',
+            ],
+            [
+                '<?php
+if ( $something ) {
+    // code
+}
+',
+                '<?php
+if ($something) {
+    // code
+}
+',
+            ],
+            [
+                '<?php
+multiply( ( 2 + 3 ) * 4 );
+',
+                '<?php
+multiply((2 + 3 ) * 4);
```

Comment:
```suggestion
multiply((2 + 3) * 4);
```
( so we see nested `)` was also covered)